cdap-common/bin/functions.sh

```diff
@@ -668,12 +670,31 @@ cdap_start_java() {
     if [[ -n ${JAVA_LIBRARY_PATH} ]]; then
       __defines+=" -Djava.library.path=${JAVA_LIBRARY_PATH}"
     fi
-    __startup_checks=${CDAP_STARTUP_CHECKS:-$(cdap_get_conf "master.startup.checks.enabled" "${CDAP_CONF}"/cdap-site.xml true)}
+    # Check if we're HDP 2.2+, otherwise, do nothing (leave up to user)
+    if [[ $(which hdp-select 2>/dev/null) ]]; then
+      local readonly __hdp_version=$(hdp-select status hadoop-client | awk '{print $3}')
+      # Check for hdp.version in OPTS
+      if [[ ${OPTS} =~ -Dhdp.version ]]; then
+        local __my_hdp_version=$(echo ${OPTS} | grep -oP '\-Dhdp.version=\d+\.\d+\.\d+\.\d+-\d*' | cut -d= -f2)
+        if [[ ${__my_hdp_version} != ${__hdp_version} ]]; then
+          logecho "[WARN] HDP version mismatch!"
+          logecho "[WARN] Detected HDP version: ${__hdp_version}"
+          logecho "[WARN] Configured HDP version: ${__my_hdp_version}"
+          echo
+          logecho "[WARN] Using configured HDP version: ${__my_hdp_version}"
```

Comment:
style-wise, should these all be one line?  Not a blocker but seems like a lot of log lines...